core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.properties
+
+import chisel3.{Bits, Element, PString, Printable, UInt}
+import chisel3.experimental.SourceInfo
+import chisel3.internal.throwException
+import chisel3.internal.firrtl.Width
+import scala.reflect.runtime.universe.{typeOf, TypeTag}
+import scala.annotation.implicitNotFound
+
+/** PropertyType defines a typeclass for valid Property types.
+  *
+  * Typeclass instances will be defined for Scala types that can be used as
+  * properties. This includes builtin Scala types as well as types defined in
+  * Chisel.
+  */
+@implicitNotFound("unsupported Property type ${T}")
```

Comment:
since i dont see it in your test, does this actually print with a substituted string without the `s"..."`? or does it just print `${T}`?

Reply:
It properly substitutes it, see the ScalaDoc for implicitNotFound! https://www.scala-lang.org/api/2.13.11/scala/annotation/implicitNotFound.html

Reply:
Yeah, I was testing it manually, but I wasn't sure how to test a compile error message in a scalatest.

Reply:
>  I wasn't sure how to test a compile error message in a scalatest.

ScalaTest doesn't work very well for that. We should probably set up a separate lit-like flow for checking compiler errors and warnings. I prototyped one a bit a while ago but just don't worry about it here.